README.md

```diff
@@ -1,29 +1,218 @@
+         "familyName",
+         "fiscalNumber",
+         "mobilePhone"
+       ],
+       name: "Required attrs"
+     },
+     spidCieUrl: "https://idserver.servizicie.interno.gov.it:8443/idp/shibboleth",
+     spidTestEnvUrl: "https://spid-testenv2:8088",
+     spidValidatorUrl: "http://localhost:8080",
+     strictResponseValidation: {
+       "http://localhost:8080": true,
+       "https://spid-testenv2:8088": true
+     }
+   };
+```
+
+* **`IDPMetadataUrl`**: URL dei metadata degli IdP. Impostare a "`https://registry.spid.gov.it/metadata/idp/spid-entities-idps.xml`".
+* **`organization`**: Oggetto con i dati del Service Provider.
+* **`publicCert`**: Stringa con il certificato del Service Provider in formato PEM.
+* **`requiredAttributes`**: La lista, in `attributes`, degli attributi richiesti
+  (identificativi in <https://docs.italia.it/italia/spid/spid-regole-tecniche/it/stabile/attributi.html>).
+* **`spidCieUrl`**: URL per l'accesso con Carta d'Identità elettronica
+  ("Entra con CIE").
+  Impostare a "`https://idserver.servizicie.interno.gov.it:8443/idp/shibboleth`"
+  per lo sviluppo.
+* **`spidTestEnvUrl`**: URL dell'istanza di [spid-testenv2](https://github.com/italia/spid-testenv2).
+  Lasciare vuoto per disabilitare.
+* **`spidValidatorUrl`**: URL dell'istanza di [spid-saml-check](https://github.com/italia/spid-saml-check).
+  Lasciare vuoto per disabilitare.
+* **`strictResponseValidation`**: Impostare come da esempio con gli URL di
+  `spid-testenv2` e `spid-saml-check` (se abilitati).
 
-- a passport-strategy that implements [SPID](https://www.spid.gov.it) authentication
-- a method that configures an express endpoint to serve Service Provider metadata
-- a scheduled procedure that refreshes IDP metadata from the [SPID registry](https://registry.spid.gov.it)
-- a redis cache provider to validate SAML `InResponseTo` field
+## Avvio dell'applicazione di esempio integrata
 
-You may use this package if you're going to implement a SPID Service Provider with a NodeJS [express server](https://expressjs.com).
+L'applicazione di esempio (`src/example.ts`) può essere lanciata con:
 
-## Upgrading passport-saml
+`docker-compose up`
 
-Beware that any changes to the method signatures of
-`SAML.prototype.generateAuthorizeRequest` and `SAML.prototype.validatePostResponse` must be reflected inside the
-[`CustomSamlClient`](./strategy/saml_client.ts) class.
+L'app scarica all'avvio i metadata di tutti gli IdP e una volta fatto (dopo il
+messaggio `[spid-express] info: samlCert expire in 12 months`), sarà
+pronta e in ascolto su <http://localhost:3000>.
 
-That's why the version of passport-saml in package.json is currently fixed at `1.2.0`.
+Iniziare la sessione SPID con una GET su `http://localhost:3000/login?entityID=ID_IDP`,
+dove `ID_IDP` è `xx_testenv2` in sviluppo.
 
-# Licence
-This repository is covered by an [MIT license](LICENSE)
+Gli id che si possono usare in produzione sono in [`src/config.ts`](src/config.ts).
```

Comment:
```suggestion
Gli idp che si possono usare in produzione sono in [`src/config.ts`](src/config.ts).
```

Reply:
I wanted to highlight the ids (fe. "lepidaid") instead of the full URLs. I'll try to rephrase that.

Reply:
:+1: